ml_data_management_code_NB.Rmd

```diff
@@ -0,0 +1,342 @@
+
+```
+
+
+### Binding data together from various cohorts / surveys
+
+Matt's very laborious way of binding data together. One note is that the binding function cannot work if the variable names differ between the datasets.
+
+``` {r include = TRUE}
+
+## Select all of the columns in the datasets to make sure they are all the same 
+## and have the same names using dplyr::select(). Then, turn them all into data tables.
+
+data1 <- as.data.table(data_a)
+data2 <- as.data.table(data_b)
+
+## Next, fix all of the variable types to make sure the variable types in each dataset
+## are aligned. For example...
+
+data1$v1_ch <- as.character(data1$v1)
+data1$v2_int <- as.integer(data1$v2)
+data2$v1_ch <- as.character(data2$v1)
+data2$v2_int <- as.integer(data2$v2)
+
+## Then, bind the datasets
+
+fulldata <- funion(data1, data2)
+
+```
+
+
+### Creating scale variables 
+
+#### Using mutate and addition
+
+``` {r include = TRUE}
+
+data <- data %>% 
+  mutate(lv1 = ((v1 + v2 + v3) / 3)) %>%
+  mutate(lv2 = ((v4 + v5 + v6) / 3)) %>%
+  mutate(lv3 = ((v7 + v8 + v9 + v10 + v11) / 5))
+
+```
+
+#### Using mutate and rowMeans
+
+``` {r include = TRUE}
+
+data <- data %>% 
+  mutate(lv1 = rowMeans(c(v1,v2,v3), na.rm = TRUE)) %>%
+  mutate(lv2 = rowMeans(c(v4,v5,v6), na.rm = TRUE)) %>%
+  mutate(lv3 = rowMeans(c(v7,v8,v9,v10,v11), na.rm = TRUE))
+
+```
+
+#### Create composite variables with a custom rowMeans function
+
+This function was used for the Clifford Harm Scale (very long scale with lots of items to create the composite)
+
+``` {r include = TRUE}
+
+  mutate(
+    clif_harm_wrong_mean = rowMeans(
+      select(., 
+             starts_with("clif_harm") & ends_with("1")
+             ), 
+             na.rm = TRUE
+      )
+
+```
+
+
+### Analyzing descriptive statistics of all items to see if there are any issue with direction or values or other factors → data checking guidelines
+
+#### Quick way to look at means and SDs
+
+``` {r include = TRUE}
+
+table <- data[c("v1", "v2", "v3", "v4", "v5")]
+
+table %>% tbl_summary(statistic = list(all_continuous() ~ "{mean} ({sd})"), 
+                       type = list(v1 ~ 'continuous',
+                                   v2 ~ 'continuous', 
+                                   v3 ~ 'continuous',
+                                   v4 ~ 'continuous',
+                                   v5 ~ 'continuous'),
+                       digits = all_continuous() ~ 2,
+                       missing_text = "(Missing)")
+
+```
+
+
+#### More detailed way to look at descriptives
+
+``` {r include = TRUE}
+
+b1<- summarySE(data, measurevar="scale", na.rm=TRUE)
+b2<- summarySE(data, measurevar="v1", na.rm=TRUE)
+b3<- summarySE(data, measurevar="v2", na.rm=TRUE)
+b4<- summarySE(data, measurevar="v3", na.rm=TRUE)
+b5<- summarySE(data, measurevar="v4", na.rm=TRUE)
+b6<- summarySE(data, measurevar="v5", na.rm=TRUE)
+
+#Now, we're going to rename the mean (the 3rd variable) to actually be called mean
+names(b1)[3] <- "mean"
```

Comment:
See above comment about applying this. I'm not quite understanding the use case.